chains/substrate/connection/check_metadata_hash_enabled_connection.go

```diff
@@ -0,0 +1,106 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package connection
+
+import (
+	"math/big"
+	"sync"
+
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/client"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/registry/parser"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/registry/retriever"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/registry/state"
+
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/rpc"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/rpc/chain"
+	"github.com/sygmaprotocol/go-substrate-rpc-client/v4/types"
+)
+
+type CheckMetadataModeEnabledConnection struct {
+	chain.Chain
+	client.Client
+	*rpc.RPC
+	meta        types.Metadata // Latest chain metadata
+	metaLock    sync.RWMutex   // Lock metadata for updates, allows concurrent reads
+	GenesisHash types.Hash     // Chain genesis hash
+}
+
+func NewCheckMetadataModeEnabledConnection(url string) (*CheckMetadataModeEnabledConnection, error) {
```

Comment:
Can we instead creating another connection just pass in the client instead so we don't copy paste connections?